psi4/src/psi4/dfocc/fno.cc

```diff
@@ -0,0 +1,263 @@
+/*
+ * @BEGIN LICENSE
+ *
+ * Psi4: an open-source quantum chemistry software package
+ *
+ * Copyright (c) 2007-2017 The Psi4 Developers.
+ *
+ * The copyrights for code used from other parties are included in
+ * the corresponding files.
+ *
+ * This file is part of Psi4.
+ *
+ * Psi4 is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, version 3.
+ *
+ * Psi4 is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License along
+ * with Psi4; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * @END LICENSE
+ */
+
+// Latest revision on October 17, 2017.
+#include <stdio.h>
+#include <iostream>
+#include <fstream>
+#include <cmath>
+#include "fno.h"
+
+namespace psi {
+namespace dfoccwave {
+
+/********************************************************************************************/
+/************************** 1d array ********************************************************/
+/********************************************************************************************/
+FnoCC::FnoCC(std::string name, int naux, int nao, int nfrzc, int naocc, int nvir, const SharedTensor2d& Corb,
+             const SharedTensor2d& Fock, const SharedTensor2d& bQ, double fno_cutoff, double fno_perct, int navir_fno,
+             bool fno_by_perct, bool fno_by_user) {
+    name_ = name;
+    naux_ = naux;
+    naob_ = nao;
+    naocc_ = naocc;
+    nfrzc_ = nfrzc;
+    nocc_ = nfrzc_ + naocc_;
+    nvir_ = nvir;
+    norb_ = nocc_ + nvir_;
+    navir_fno_ = navir_fno;
+    fno_cutoff_ = fno_cutoff;
+    fno_percentage_ = fno_perct;
+    fno_by_perct_ = fno_by_perct;
+    fno_by_user_ = fno_by_user;
+    cutoff_ = 1.0E-10;
+
+    // malloc
+    Corb_ = SharedTensor2d(new Tensor2d("FNO Alpha MO Coefficients", naob_, norb_));
```

Comment:
Make sure to change all of these to `std::make_shared<Tensor2d>("FNO Alpha MO Coefficients", naob_, norb_)`.